index.html

```diff
@@ -3,69 +3,103 @@
+            height: 100%;
+            overflow: hidden;
+        }
+        #viewer-container {
+            position: relative;
             width: 100%;
-            height: 600px;
+            height: 100%;
         }
-        .image-buttons {
-            margin: 10px 0;
+        #viewer {
+            position: absolute;
+            height: 100%;
+            width: 100%;
         }
-        .image-buttons button {
-            margin-right: 10px;
+        #divider {
+            position: absolute;
+            top: 0;
+            bottom: 0;
+            width: 2px;
+            background: black;
+            pointer-events: none;
         }
     </style>
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/OpenSeadragonScalebar/2.0.0/openseadragon-scalebar.min.css" />
 </head>
 <body>
-    <div class="image-buttons">
-        <button data-image="images/O48-013.dzi">XPL</button>
-        <button data-image="images/O48-014.dzi">PPL</button>
+    <div id="viewer-container">
+        <div id="viewer"></div>
+        <div id="divider"></div>
     </div>
-    <div id="openseadragon"></div>
     <script src="js/openseadragon.min.js"></script>
     <script src="js/openseadragon-scalebar.min.js"></script>
     <script>
-        var viewer = OpenSeadragon({
-            id: "openseadragon",
+        const viewer = OpenSeadragon({
+            id: "viewer",
             prefixUrl: "js/images/",
-            tileSources: "https://raw.githubusercontent.com/grsharman/petro-image/main/images/O48-013.dzi"
+            tileSources: [
+                "https://raw.githubusercontent.com/grsharman/petro-image/main/images/O48-013.dzi",
+                "https://raw.githubusercontent.com/grsharman/petro-image/main/images/O48-014.dzi",
+            ]
+        });
+
+        const divider = document.getElementById('divider');
+        const viewerContainer = document.getElementById('viewer-container');
+
+        // Keep track of the mouse's position outside the pointermove event so
+        // that it's available when handling other events, such as viewer
+        // animation.
+        let mousePos = new OpenSeadragon.Point(0, 0);
+
+        // Sets the dividing line between the two images at the given
+        // x-coordinate in window space.
+        const divideAtX = x => {
+            // Reposition the divider between the partial images.
+            divider.style.left = x + 'px';
+
+            // Clip the overlayed image to show only the region to the right of
+            // x, exposing the image underneath.
+            const overlay = viewer.world.getItemAt(1);
+            const clipPos = overlay.viewerElementToImageCoordinates(new OpenSeadragon.Point(x, 0));
+            // Clamp the clip to within the image bounds.
+            const size = overlay.getContentSize();
+            const xClip = Math.max(0, Math.min(clipPos.x, size.x));
+            overlay.setClip(new OpenSeadragon.Rect(xClip, 0, size.x, size.y));
```

Comment:
For some reason I couldn't figure out, this call generates assertion failures within OpenSeaDragon when the image is both entirely out of view and has a small enough clip. This behavior can be triggered by dragging and holding the image far to the left. There doesn't appear to be any degraded behavior - it just generates log noise if the user happens to be in the developer tools.